### PR TITLE
Fix: chain suggestions

### PR DESCRIPTION
**Pull request by DavideSegullo**
Chain Suggestions compiled by our frontend are faulty and not showing balances within the wallet. (For example Passage config is broken on production, balance doesn't load inside Keplr).

This PR fixes the list of assets we pass to the `WalletManager` class to make sure we provide it with the native data of the chain it belongs to and not the IBC data.

### ClickUp Task

[ClickUp Task URL](https://app.clickup.com/t/86a1xqh15)

### CODE REVIEW

File: packages/web/stores/root.ts [hunk 1]
```diff
@@ -55,6 +55,29 @@ import {
...
+    };
+  });
+
+  return {
+    ...assetList,
+    assets,
+  };
+});
+
+console.log(RawAssetLists.find((el) => el.chain_id === "passage-2"));
```
Comment (MaxMillington) [inline packages/web/stores/root.ts:79]: can we remove the log

**Review by MaxMillington — APPROVED** [review]
let's get jeremy to QA

File: packages/web/stores/root.ts [hunk 2]
```diff
@@ -55,6 +55,29 @@ import {
...
+      ...asset,
+      base,
+    };
+  });
+
+  return {
+    ...assetList,
+    assets,
+  };
+});
```
Comment (JoseRFelix) [inline packages/web/stores/root.ts:77]: I believe we shouldn't modify the assetlist at the root store level. Instead, how about we move this logic into the account store?

File: packages/web/stores/root.ts (hunk 1 above)
Comment (DavideSegullo) [reply packages/web/stores/root.ts]: Done

File: packages/web/stores/root.ts [hunk 3]
```diff
@@ -55,6 +55,27 @@ import {
...
+    let base = asset.base;
+
+    if (ibcTrace) {
+      base = ibcTrace.counterparty.base_denom;
+    }
+
+    return {
+      ...asset,
+      base,
+    };
```
Comment (JoseRFelix) [inline packages/web/stores/root.ts:70]: Btw, we can get the source denom using the [`getSourceDenomFromAssetList`](https://github.com/osmosis-labs/osmosis-frontend/blob/e6fb7743580039d926d1448df0070506f4f34c5b/packages/utils/src/asset-utils.ts#L18) function in @osmosis-labs/utils.

**Review by JoseRFelix — CHANGES_REQUESTED** [review]

File: packages/web/stores/root.ts (hunk 3 above)
Comment (DavideSegullo) [reply packages/web/stores/root.ts]: Done

File: packages/web/stores/root.ts (hunk 2 above)
Comment (DavideSegullo) [reply packages/web/stores/root.ts]: You could do it, in general I wouldn't do it for a generic class that tends to be controlled by a 'parent,' but in this case it's to make sure the data is in the correct format, so I would say yes

Comment (DavideSegullo) [reply packages/web/stores/root.ts]: Done

**Comment by CryptoAssassin1** [comment]
Fixed.
Screenshot from 2024-01-13 01-35-42

**Review by CryptoAssassin1 — APPROVED** [review]

**Comment by JoseRFelix** [comment]
@DavideSegullo Looks like tests are failing

**Comment by DavideSegullo** [comment]
> @DavideSegullo Looks like tests are failing

Fixed

File: packages/stores/src/account/base.ts [hunk 4]
```diff
@@ -126,6 +131,20 @@ export class AccountStore<Injects extends Record<string, any>[] = []> {
 
   private _cache = new LRUCache<string, CacheEntry>({ max: 30 });
 
+  /**
+   * We make sure that the 'base' field always has as its value the native chain parameter
+   * and not values derived from the IBC connection with Osmosis
+   */
+  private get _assets() {
```
Comment (JoseRFelix) [inline packages/stores/src/account/base.ts:138]: What are your thoughts on renaming this to `walletManagerAssets` since we only need this for cosmos-kit?

Comment (DavideSegullo) [reply packages/stores/src/account/base.ts:138]: Yes makes sense

Comment (DavideSegullo) [reply packages/stores/src/account/base.ts]: Done

**Review by JoseRFelix — APPROVED** [review]

**Review by kamal-sutra — APPROVED** [review]

**Comment by JeremyParish69** [comment]
FWIW: the Passage Chain Suggestion has now been added to Keplr's Chain Registry, so that would override whatever we suggested. But I was able to eventually find that Lambda chain is not in their registry, and I could see that the compiled Chain Suggestion correctly uses LAMB for its currency instead of Osmosis' ibc hash denom. So I think this fixes it!

**Pull request merged**